### ISSUE TITLE
🐛 Biome issues warnings on certain files produced by `supervise`

### ISSUE THREAD

**Issue by MCFX2** [biomejs/biome#2311]
### Environment information

```block
CLI:
  Version:                      1.6.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.12.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/4.1.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

1. Run `supervise .` at the root of a Node project, and optionally kill it afterwards with ctrl+C
2. Note that this utility creates a `supervise` folder with some files, such as `supervise/control` and `supervise/ok`
3. Make sure that `supervise/*` is added to linter ignore list (I tried a few variations, `supervise/**` `supervise/**/*` `supervise/*.*` with same result)
4. Run `biome lint .`
5. below message is reported on files in the `supervise` folder

```
./supervise/ok internalError/fs ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ⚠ Unknown file type

  ℹ Biome encountered a file system entry that's neither a file, directory or symbolic link

```

### Expected result

There should be no output on these files, especially since they match a pattern in the ignore list.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by MCFX2** [comment]
Related: [Documentation on the `supervise` utility](https://manpages.ubuntu.com/manpages/focal/en/man8/supervise.8.html).

**Comment by arendjr** [comment]
Does it help if you put the `supervise` directory itself in the ignore list? Or `**/supervise`? I agree your solution should also work, but this may be a workaround.

**Comment by MCFX2** [comment]
> Does it help if you put the supervise directory itself in the ignore list? Or **/supervise? I agree your solution should also work, but this may be a workaround.

`"supervise/"`, `"**/supervise"`, `"supervise"`, `"**/supervise/"`, `"./supervise"`, and `"./supervise/"` all produce the same results.

**Comment by MCFX2** [comment]
After some local investigation, the files it's tripping up on are [named pipes created with something like `mkfifo`](https://stackoverflow.com/questions/4113986/example-of-using-named-pipes-in-linux-shell-bash), which Biome doesn't seem too fond of.

So the issues at stake here are:
- Biome probably shouldn't consider a named pipe an "unknown file type", and should ignore all named pipes.
- Biome should not be traversing directories that are matched by an ignore pattern.
- (Opinion) Biome encountering an unknown file type should not issue a warning.

**Comment by MCFX2** [comment]
Looks like Biome also fails in the same way on sockets (which similarly can appear in the filesystem on Linux systems).

**Comment by Sec-ant** [comment]
Just for reference, rust has unix-specific extensions for the fs module that is able to detect fifo and socket files: https://doc.rust-lang.org/std/os/unix/fs/trait.FileTypeExt.html
